app/modules/vm/index.coffee

```diff
@@ -472,11 +475,17 @@ module.exports = angular.module 'xoWebApp.vm', [
 
         return
 
+      # Disk resize
+      forEach disks, (attributes, id) ->
+        disk = get id
+        if attributes.size isnt disk.size
+          promises.push (xo.disk.resize id, attributes.size)
+
       forEach disks, (attributes, id) ->
         # Keep only changed attributes.
         disk = get id
         forEach attributes, (value, name) ->
-          delete attributes[name] if value is disk[name]
+          delete attributes[name] if value is disk[name] or name is 'size'
```

Comment:
Indeed, my bad, perhaps it would be better delete this attribute when we do the disk resize.